### PR TITLE
Replace rspec mocked out unit tests with minitest integration tests.

### PR DESCRIPTION
**Pull request by dylanahsmith**
@EiNSTeiN- & @clayton-shopify for review

## Problem

Commit 4adfbadd1a2c6b69564459f35c91765d2a46233d (v1.1.2) started validating the site in both the authorization and callback endpoints, but pull request #11 caused a regression (in v1.1.3) where we stopped validating the site in the callback endpoint (where the access token is sent).   This wasn't fixed until 0d191f4dfb3e67f5e46281272a368e1d5294483c (v1.1.8), yet we were still lacking a proper regression test.

## Solution

The rspec tests that we had were full of mocks, and unit tests for very specific methods.  What we need is integration tests to make sure everything actually works together.  So this pull request replaces the entire test suite with as close to integration tests as convenient.  The only mocking is through fakeweb and rack mocked requests, so no ad hoc stubbing.

I chose to use minitest since I was already replacing the test suite, and because most Shopify devs are more familiar with minitest.

I tried testing with commit 0d191f4dfb3e67f5e46281272a368e1d5294483c reverted, and confirmed the regression test catches that bug.

### CODE REVIEW

File: test/integration_test.rb [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+  def teardown
+    FakeWeb.clean_registry
+    FakeWeb.last_request = nil
+  end
+
+  def test_authorize
+    response = authorize('snowdevil.myshopify.com')
+    assert_equal 302, response.status
+    assert_match /\A#{Regexp.quote("https://snowdevil.myshopify.com/admin/oauth/authorize?")}/, response.location
```
Comment (EiNSTeiN-) [inline test/integration_test.rb]: [according to this](http://api.rubyonrails.org/classes/ActionDispatch/Assertions/ResponseAssertions.html) you could do:
```
assert_redirected_to /\A#{Regexp.quote("https://snowdevil.myshopify.com/admin/oauth/authorize?")}/
```

Comment (EiNSTeiN-) [inline test/integration_test.rb]: Nevermind, minitest does not have `assert_redirected_to` :(

**Comment by EiNSTeiN-** [comment]
:+1:

**Comment by clayton-shopify** [comment]
:+1:

**Pull request merged**